machine/src/lib.rs

```diff
@@ -119,7 +138,7 @@ impl<F> Into<[F; MEMORY_CELL_BYTES]> for Word<F> {
 
 pub trait Machine {
     type F: Field;
-    fn run(&mut self);
+    fn run(&mut self, initial_state: ProgramState<Self::F>, program: ProgramROM<Self::F>);
```

Comment:
nit: could the impl do `ProgramState::default()` so we don't need to pass it?

Reply:
I think we need to pass the initial register state to a run, as program execution will depend on initial `pc`. Would you prefer if these registers are passed directly as arguments?